### PR TITLE
Remove promise-to-callback, reimplement createAsyncMiddleware

### PR DESCRIPTION
**Pull request by rekmarks**
Removes `promise-to-callback`, and reimplements `createAsyncMiddleware` using Promises and `async`/`await`.

`promise-to-callback` is a really tiny (<1kb) dependency, but the new implementation is more idiomatic, better documented, better covered by existing tests, and IMO overall easier to understand.

_Note:_ **Undefined** behavior happens if an `async` middleware calls `next` more than once, but it appears to blow up in the same way with this implementation as in `5.1.8`.

### CODE REVIEW

File: src/createAsyncMiddleware.js [hunk 1]
```diff
@@ -1,39 +1,59 @@
-const promiseToCallback = require('promise-to-callback')
+/**
+ * JsonRpcEngine only accepts synchronous middlewares.
```
Comment (Gudahtt) [inline src/createAsyncMiddleware.js]: Same notes here as in the README; "middleware" is already plural, and callback-based middleware is still asynchronous.

**Review by Gudahtt — COMMENTED** [review]
This implementation looks correct to me 👍

File: src/createAsyncMiddleware.js (hunk 1 above)
Comment (rekmarks) [reply src/createAsyncMiddleware.js]: Fixed in: 1448044

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**